src/Schedule.h

```diff
@@ -118,28 +119,34 @@ struct Split {
                 // tail strategy to be GuardWithIf.
     TailStrategy tail;
 
-    enum SplitType {SplitVar = 0, RenameVar, FuseVars};
+    enum SplitType {SplitVar = 0, RenameVar, FuseVars, PurifyRVar};
 
     // If split_type is Rename, then this is just a renaming of the
     // old_var to the outer and not a split. The inner var should
     // be ignored, and factor should be one. Renames are kept in
     // the same list as splits so that ordering between them is
     // respected.
 
+    // If split type is Purify, this replaces the old_var RVar to
+    // the outer Var. The inner var should be ignored, and factor
+    // should be one.
+
     // If split_type is Fuse, then this does the opposite of a
     // split, it joins the outer and inner into the old_var.
     SplitType split_type;
 
     bool is_rename() const {return split_type == RenameVar;}
     bool is_split() const {return split_type == SplitVar;}
     bool is_fuse() const {return split_type == FuseVars;}
+    bool is_purify() const {return split_type == PurifyRVar;}
 };
 
 struct Dim {
     std::string var;
     ForType for_type;
     DeviceAPI device_api;
     bool pure;
```

Comment:
I don't think you can have an impure non-rvar. Instead of two bools, it may be better to have an enum {PureVar, PureRVar, ImpureRVar}